src/tyr/isochrone_serializer.cc

```diff
@@ -22,12 +22,15 @@ template <class coord_t>
 std::string
 serializeIsochrones(const Api& request,
                     const typename midgard::GriddedData<coord_t>::contours_t& grid_contours,
+                    const std::vector<valhalla::LatLng>& snapped_centers,
                     bool polygons,
                     const std::unordered_map<float, std::string>& colors,
                     bool show_locations) {
   // for each contour interval
   int i = 0;
   auto features = array({});
+  // no support for multiple locations per request currently
+  valhalla::LatLng snapped_center = snapped_centers.at(0);
```

Comment:
instead of this you can do

`request.options().locations(0).path_edges(0).ll()`

which will allow you to remove pretty much all the other code in this pr expect the serialization below.

we do have another problem though. the api supports multiple locations and we dont know to which contour a center belongs. whats worse is that, the way that we build the contours we loose all ability to do that anyway...

in my humble opinion i think we should add a new parameter to the api, that also serializes another geojson feature with geometry type MultiPoint and from there we can add all the candidate points we used to make the isochrones. @nilsnolde what would you think of that?

Reply:
> request.options().locations(0).path_edges(0).ll()

> which will allow you to remove pretty much all the other code in this pr except the serialization below.

Oh jeez, really :sweat_smile: Thanks for the pointer, so easy..

> the api supports multiple locations

Yeah, David told me how that's supposed to work. However, that seems to be a slightly broken feature atm (IMHO), see #2115 . You think that needs to be addressed before or in general @kevinkreiser ?

> we should add a new parameter to the api, that also serializes another geojson feature with geometry type MultiPoint 

I like that better as well, then those points can be rendered directly from the FeatureCollection! I guess you don't want to hijack `show_locations` for this purpose? Rather add a new parameter, like `snapped_locations`?

Reply:
I think hijiacking show_locations is fine. we just need a property that distinguishes input point from snapped point and i think we would be good to go!